tests/test_frontdoor.py

```diff
@@ -302,6 +302,224 @@ def test_back_button():
+
+    form = page.get_form()
+    form["building_name_or_number"] = "10"
+    form["postcode"] = "SW1A 2AA"
+    page = form.submit().follow()
+
+    data = interface.api.session.get_answer(session_id, page_name="address")
+    assert data["building_name_or_number"] == "10"
+    assert data["postcode"] == "SW1A 2AA"
+
+    form = page.get_form()
+    form["uprn"] = "100023336956"
+    page = form.submit().follow()
+
+    data = interface.api.session.get_answer(session_id, page_name="address-select")
+    assert data["uprn"] == "100023336956"
+    assert data["address"] == "10, DOWNING STREET, LONDON, CITY OF WESTMINSTER, SW1A 2AA"
+
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.has_one("h1:contains('Is anyone in your household receiving any of the following benefits?')")
+
+    page = page.click(contains="Back")
+
+    assert page.has_one('h1:contains("What is the property\'s address?")')
+
+
+@unittest.mock.patch("help_to_heat.frontdoor.interface.EPCApi", MockNotFoundEPCApi)
+@unittest.mock.patch("help_to_heat.frontdoor.interface.OSApi", MockOSApi)
+@utils.mock_os_api
+def test_property_type_back_button_with_social_housing_and_no_epc():
+    client = utils.get_client()
+    page = client.get("/start")
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.status_code == 200
+    session_id = page.path.split("/")[1]
+    assert uuid.UUID(session_id)
+    _check_page = _make_check_page(session_id)
+
+    form = page.get_form()
+    form["country"] = "England"
+    page = form.submit().follow()
+
+    form = page.get_form()
+    form["supplier"] = "Utilita"
+    page = form.submit().follow()
+
+    assert page.has_text("Do you own the property?")
+    page = _check_page(page, "own-property", "own_property", "No, I am a social housing tenant")
+
+    form = page.get_form()
+    form["building_name_or_number"] = "10"
+    form["postcode"] = "SW1A 2AA"
+    page = form.submit().follow()
+
+    data = interface.api.session.get_answer(session_id, page_name="address")
+    assert data["building_name_or_number"] == "10"
+    assert data["postcode"] == "SW1A 2AA"
+
+    form = page.get_form()
+    form["uprn"] = "100023336956"
+    page = form.submit().follow()
+
+    data = interface.api.session.get_answer(session_id, page_name="address-select")
+    assert data["uprn"] == "100023336956"
+    assert data["address"] == "10, DOWNING STREET, LONDON, CITY OF WESTMINSTER, SW1A 2AA"
+
+    assert page.status_code == 302
+    page = page.follow()
+
+    assert page.has_one("h1:contains('What kind of property do you have?')")
+
+    page = page.click(contains="Back")
+
+    assert page.has_one('h1:contains("What is the property\'s address?")')
+
+
+@unittest.mock.patch("help_to_heat.frontdoor.interface.OSApi", MockOSApi)
+@utils.mock_os_api
+def test_property_type_back_button_with_social_housing_and_scotland():
+    client = utils.get_client()
```

Comment:
Only if it ends up being really quick, otherwise we can leave it for next time:

There's a fair bit of common steps between these tests. One of the things I'm keen to do with this file is extract duplicated logic to helper methods like `fill_in_such_and_such_a_page/pages`. I'm hoping we can chip away at is from time to time as we go. Would it be easy to do something like that here?

Reply:
Are you envisioning chunky helpers with several questions answered, or smaller ones like:

`def country_selection(page: interface.api.session, country: str) -> interface.api.session:
    form = page.get_form()
    form["country"] = country
    return form.submit().follow()
`

Reply:
As discussed, something more like the last thing. If you've already got something in that direction then do throw it up here and I'll have a look, but I think this is good to merge as-is now.

Reply:
The bit I've put together isn't working initially, so might be best to just merge as is now. 
But I've got some ideas now for if we come back to this.